### PR TITLE
API cleanup.

### PR DESCRIPTION
**Pull request by feuerste**
This PR cleans up the interface by hiding non-public functions.

### CODE REVIEW

File: xray/src/generation.rs [hunk 1]
```diff
@@ -545,6 +543,7 @@ pub fn build_xray_quadtree(
     let (all_nodes_tx, all_nodes_rx) = mpsc::channel();
     println!("Building level {}.", deepest_level);
 
+    let global_from_local = &local_from_global.clone().map(|t| t.inverse());
```
Comment (nnmm) [inline xray/src/generation.rs]: Naive question: Why is this clone() needed?

Comment (feuerste) [reply xray/src/generation.rs]: We create a reference to a temporary here, without clone this wouldn't be possible.

Comment (nnmm) [reply xray/src/generation.rs]: What do you think about using `as_ref()`? Not that performance matters here, but it feels slightly cleaner.

Comment (feuerste) [reply xray/src/generation.rs]: Yep, good idea, done.

**Comment by nnmm** [comment]
:100:

**Comment by feuerste** [comment]
@nnmm I think here a :100: unfortunatly isn't enough to approve :disappointed:

**Comment by feuerste** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user feuerste. Merge queue now has a length of 1.

**Pull request merged**